src/components/dialogs/inputs/reactive-capability-curve-table.js

```diff
@@ -33,6 +33,44 @@ function hasValueCorrectFormat(valueToTest, index) {
     );
 }
 
+function getPreviousValue(displayedPreviousValues, displayedValues, index) {
+    if (
+        displayedPreviousValues !== undefined &&
+        displayedValues &&
+        index === displayedValues.length - 1
+    ) {
+        return {
+            p: displayedPreviousValues[displayedPreviousValues.length - 1].p,
+            qminP: displayedPreviousValues[displayedPreviousValues.length - 1]
+                .qminP,
+            qmaxP: displayedPreviousValues[displayedPreviousValues.length - 1]
+                .qmaxP,
+        };
+    } else if (
+        displayedPreviousValues !== undefined &&
+        displayedPreviousValues[index] &&
+        index < displayedPreviousValues.length - 1
+    ) {
+        return {
+            p: displayedPreviousValues[index].p,
+            qminP: displayedPreviousValues[index].qminP,
+            qmaxP: displayedPreviousValues[index].qmaxP,
+        };
+    } else {
+        return null;
+    }
+}
+
+function fixValuesFormat(valuesToFix) {
```

Comment:
This function is bad. Why is it here ?
There is a bug if the value test is equal to zero. There are validation functions available for the job in the validation-functions.js file if needed.